### PR TITLE
Data preparation

### PR DESCRIPTION
**Pull request by nprisbrey**
Okay folks! This is the first version of data preparation that is ready to be tested. Can @mheilner @Hcorry98 @gavinbolin look this over and see if there's any obvious issues?

The code basically reads in Yahoo Finance CSVs, cleans the data, and loads it into PyTorch Datasets. An example of loading them into Dataloaders:

```
import torch
from load_data import get_data_tensor, get_train_test_datasets
from torch.utils.data import DataLoader

all_data = get_data_tensor()
train_dataset, test_dataset = get_train_test_datasets(all_data, 7)

train_dataloader = DataLoader(train_dataset, batch_size=3, shuffle=True)
test_dataloader = DataLoader(test_dataset, batch_size=3, shuffle=True)
```

I say we push this through and I'll iron out any bugs that come up as we continue with the project. Everything currently runs smoothly on my end.

### CODE REVIEW

**Comment by mheilner** [comment]
Looked through it and at first glance it looks great. The best part is the scalability with other features. We will talk more about this tonight

**Review by mheilner — APPROVED** [review]

File: load_data.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    new_col_order.remove(target_column)
+    master_df = master_df[[target_column] + new_col_order]
+
+    # Convert master_df to a standardized, combined PyTorch tensor
+    master_tens = torch.from_numpy(master_df.to_numpy(dtype=np.float32))
+
+    # If relative_change, convert data to fraction of change from previous day
+    if relative_change:
+        master_tens_rd = master_tens.roll(1, 0)
+        master_tens = (master_tens - master_tens_rd) / torch.abs(master_tens_rd)
```
Comment (mheilner) [inline load_data.py:176]: remove the absolute target when this is input

**Review by gavinbolin — APPROVED** [review]
Looks awesome! Great job Nathan

**Comment by gavinbolin** [comment]
Added TestDataset.py to test various types of learning models. Implemented only the Perceptron for now. Edited Stock Dataset as it was returning a StockDataset item from the __getitem__ func which was complicating getting the tensor item. load_data.py was returning a tensor alone from the get_data function rather than a StockDataset object. It might be easier to just return a tensor when the only other value that it has is the seq_len which we can store at initialization.

**Pull request merged**